### PR TITLE
[Node finder] make farm name and location fields as autocomplete fields

### PR DESCRIPTION
**Pull request by 0oM4R**
### Description

using `TFSelectLocation` and `TFSelectFarm` components to provide searching and auto completion;  

### Changes

- in `TfSelectLocation.vue` we added a prop `NodeStatus` to get the locations based on the provided node status
 - This value is used in getLocations function, so in solutions <[TfSelectionDetails.vue](https://github.com/threefoldtech/tfgrid-sdk-ts/compare/development_autoCompleteFilters?expand=1#diff-b11587292fa1b7595af2337c33cb564c177389b82ed451a4ef607d955890275a)> we are passing this prop as `NodeStatus.Up` to get only up nodes,  but in filters we need to get all locations whatever the node status is, 
- added a slot in filters to prepend the region, country and farm name fields
- the rest of changes is related to the now fields added

image

> this pr is only for node finder will create another one for Farm Finder
### Related Issues

- #1901 
- #1902
- #1869

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Review by MohamedElmdary — APPROVED** [review]

File: packages/playground/src/components/node_selector/TfSelectLocation.vue
Comment (Mahmoud-Emad) [inline packages/playground/src/components/node_selector/TfSelectLocation.vue:1]: Although Vue conventionally follows PascalCase for naming components, we have used snake_case in our project, do you think we need to change the file name in this PR?

File: packages/playground/src/utils/filter_nodes.ts [hunk 1]
```diff
@@ -48,45 +45,30 @@ export type DedicatedNodeFilters = {
...
-  farmIds: {
-    label: "Farm ID",
+  nodeId: {
+    label: "Node ID",
     rules: [
       [
         isNumeric("This field accepts numbers only.", { no_symbols: true }),
-        min("The ID should be larger than zero.", 1),
-        isInt("should be an integer"),
+        min("The node id should be larger then zero.", 1),
```
Comment (Mahmoud-Emad) [inline packages/playground/src/utils/filter_nodes.ts:66]:
```suggestion
        min("The node ID should be larger then zero.", 1),
```

File: packages/playground/src/utils/filter_nodes.ts [hunk 2]
```diff
@@ -48,45 +45,30 @@ export type DedicatedNodeFilters = {
...
+        startsWith("The node id start with zero.", "0"),
```
Comment (Mahmoud-Emad) [inline packages/playground/src/utils/filter_nodes.ts:67]:
```suggestion
        startsWith("The node ID start with zero.", "0"),
```

File: packages/playground/src/utils/filter_nodes.ts [hunk 3]
```diff
@@ -48,45 +45,30 @@ export type DedicatedNodeFilters = {
...
-  },
-  country: {
-    label: "Country Full Name",
-    type: "text",
-    tooltip: "Filter by country.",
-  },
-  region: {
-    label: "Region Name",
-    tooltip: "Filter by region.",
+    tooltip: "Filter by node id.",
```
Comment (Mahmoud-Emad) [inline packages/playground/src/utils/filter_nodes.ts:71]:
```suggestion
    tooltip: "Filter by node ID.",
```

File: packages/playground/src/utils/get_nodes.ts [hunk 4]
```diff
@@ -178,9 +178,9 @@ export const getQueries = (mixedFilters: MixedFilter): Partial<NodesQuery> => {
     retCount: true,
     nodeId: +mixedFilters.inputs.nodeId.value! || undefined,
     farmIds: mixedFilters.inputs.farmIds.value,
-    farmName: mixedFilters.inputs.farmName.value,
-    country: mixedFilters.inputs.country.value,
-    region: mixedFilters.inputs.region.value,
+    farmName: mixedFilters.options.farmName,
+    country: mixedFilters.options.country,
+    region: mixedFilters.options.region,
```
Comment (Mahmoud-Emad) [inline packages/playground/src/utils/get_nodes.ts:183]: Why was it removed from the inputs and added to the options?

The distinction between `inputs` and `options` is as follows:

- Inputs: Any input provided by the user, such as `input`, `select`, `autocomplete`, and so on.

- Options: Filters that can be applied to the inputs, such as the page number, table count, switches, and so on.

File: packages/playground/src/views/nodes.vue [hunk 5]
```diff
@@ -119,10 +168,10 @@ export default {
     const nodesCount = ref<number>(0);
     const filtering = ref(false);
     const selectedNodeId = ref<number>(0);
-
+    const farm = ref();
     const isDialogOpened = ref<boolean>(false);
     const isValidForm = ref<boolean>(true);
-
+    const location = ref();
```
Comment (Mahmoud-Emad) [inline packages/playground/src/views/nodes.vue:174]: Please indicate the type of `farm` and `location` in the respective fields.

**Comment by Mahmoud-Emad** [comment]
I think there is something unusual in this behavior. Can you please check this video?

  [Screencast from 01-15-2024 11:26:44 AM.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/57001890/b1623f95-7032-49b6-bbc3-f9ccfbeea67e)

The query params are used to save the initial state and compare it to the current state.

**Pull request closed without merging**